packages/protocol/thread/thread.ts

```diff
@@ -883,6 +899,23 @@ class _ThreadFront {
     );
   }
 
+  findMessagesInRange(range: PointRange) {
+    return client.Console.findMessagesInRange({ range }, ThreadFront.sessionId!);
+  }
+
+  async findKeyboardEvents(onKeyboardEvents: (events: keyboardEvents) => void) {
+    const keyboardEventsLoaded = client.Session.findKeyboardEvents({}, this.sessionId!);
+    client.Session.addKeyboardEventsListener(onKeyboardEvents);
+    return keyboardEventsLoaded
+  }
+
+  async findNavigationEvents(onNavigationEvents: (events: navigationEvents) => void) {
+    const navigationEventsLoaded = client.Session.findNavigationEvents({}, this.sessionId!);
+    client.Session.addNavigationEventsListener(onNavigationEvents);
+    return navigationEventsLoaded
```

Comment:
Same as above (add the handler before calling `findNavigationEvents`)
```suggestion
    client.Session.addNavigationEventsListener(onNavigationEvents);
    return client.Session.findNavigationEvents({}, this.sessionId!);
```

Reply:
good call